### ISSUE TITLE
RuntimeError: CUDA error: invalid device function

### ISSUE THREAD

**Issue by hadaev8** [NVIDIA/waveglow#161]
Tried to run it on google colab as usual and got this error.
So it dont like .half() function?

Trace:
```
/content/tacotron2/waveglow/glow.py in infer(self, spect, sigma)
    281             audio = torch.cat([audio_0, audio_1],1)
    282 
--> 283             audio = self.convinv[k](audio, reverse=True)
    284 
    285             if k % self.n_early_every == 0 and k > 0:

/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py in __call__(self, *input, **kwargs)
    545             result = self._slow_forward(*input, **kwargs)
    546         else:
--> 547             result = self.forward(*input, **kwargs)
    548         for hook in self._forward_hooks.values():
    549             hook_result = hook(self, input, result)

/content/tacotron2/waveglow/glow.py in forward(self, z, reverse)
     92                 W_inverse = Variable(W_inverse[..., None])
     93                 if z.type() == 'torch.cuda.HalfTensor':
---> 94                     W_inverse = W_inverse.half()
     95                 self.W_inverse = W_inverse
     96             z = F.conv1d(z, self.W_inverse, bias=None, stride=1, padding=0)

RuntimeError: CUDA error: invalid device function
```

**Comment by hadaev8** [comment]
Link
https://colab.research.google.com/drive/1i72zYqaSub9iAvWTyVWVyTeVNRZ_XNlY

**Comment by rafaelvalle** [comment]
It looks like you're running this on the CPU.

**Comment by zm379096736** [comment]
How do you solve it?

**Comment by hadaev8** [comment]
reverted pytorch version

**Comment by zm379096736** [comment]
> reverted pytorch version

thanks, wrong version of cudatoolkit during installation for pytorch.

**Comment by giuliopaci** [comment]
Hi I am experiencing the same issue while trying to run https://github.com/NVIDIA/tacotron2/blob/master/inference.ipynb on my PC.

I am using Pytorch 1.2.0, with Cuda 9.2, installed trough `pip3 install --user torch==1.2.0+cu92 torchvision==0.4.0+cu92 -f https://download.pytorch.org/whl/torch_stable.html`.

Which version of Pytorch should I use instead? Is 1.0 version mandatory?

**Comment by giuliopaci** [comment]
For anybody interested... I noticed that pytorch developers claimed this issue to be solved in 1.3.1. However I was not able to use that version, because precompiled versions of pytorch dropped support for my GPU.

In the end I was able to proceed by:
- uninstalling torch and torchvision;
- compile magma, torch, apex from source;

I compiled magma, torch, apex starting from:
1. https://bitbucket.org/icl/magma.git commit `23f549e9a238ca51710eadae1ff6a9d840f09aed`
2. https://github.com/pytorch/pytorch commit `4ed7f36ed181ee784f9904d5eabf073701e1fb78`
3. https://www.github.com/nvidia/apex commit `8a1ed9e8d35dfad26fb973996319965e4224dcdd`

I was able to correctly perform inference (I did not try training).
